testsuite/compare.cmake

```diff
@@ -0,0 +1,27 @@
+# This is used to compare to files using CMake.
+# It improves on the default behaviour to print the
+# the files if there is a difference.
+
+execute_process(
+    COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol ${original_file} ${new_file}
+    RESULT_VARIABLE status
+)
+
+if (${status} EQUAL 1)
+    message ("Compare ${original_file} with ${new_file}")
+    file(READ ${original_file} original_text)
+    file(READ ${new_file} new_text)
+    message("--Original File-----------------------------------------------------------------")
+    if (NOT original_text STREQUAL "")

```

Comment:
Yes.  This is going to present everything.  My current approach would be just to run
```
   ninja update-dump
```
and then 
```
   git diff
```
That could be done in CI. To give a contextual output.  @plietar was not keen for update-dump to be the only approach.  So I implemented a `ctest` approach, and both could be used.

Reply:
right, `git diff` could do it better. Then why do we need the actual dump during the tests? We could just have a `message` at the end, if error, to do a `git diff`, no?

Reply:
I don't know how to do that in CI in that way, the output is primarily for debugging in CI.  Also, you might not want to perturb your checkout in that way.  It outputs the two files to run diff on too.

Reply:
To run the diff on the output you'd have to copy and paste into two files and diff, no? If there are multiple (similar) errors (on different tests), than this would be a very cumbersome way to investigate errors.

It may sound hacky, but if we could check for a command line called `diff` and use it if available, and if not, just dump the files, this would at least work towards something more amenable to some developers?

Reply:
> It may sound hacky, but if we could check for a command line called diff and use it if available, and if not, just dump the files, this would at least work towards something more amenable to some developers?

I like this approach, I'll do that.  Hopefully, we will get nicer behaviour for almost everyone.

Reply:
Done.  You should be able to replace the diff tool with setting the following for cmake
```
-DDIFF_TOOL=myfavouritediff
```
If you don't set that, then it searches for `diff`, if it fails to find `diff`, then it prints both files.

I amended Windows CI to install `diff`. So hopefully failures we be easier to debug.

Reply:
Awesome!